tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/page/PageReader.java

```diff
@@ -162,7 +163,11 @@ public Statistics getStatistics() {
 
   @Override
   public void setFilter(Filter filter) {
-    this.filter = filter;
+    if (this.filter == null) {
```

Comment:
@JackieTien97 check whether this has some negative effect.

Reply:
Actually, i think we should just use the filter to overwrite the former one. The former one is actually a time filter and it has been checked when the PageReader is constructed

Reply:
I previously found that data of this page will miss the previous filter, meaning some data are not time-filtered. We may remove it later if you are confident.